src/libstd/ffi/c_str.rs

```diff
@@ -444,6 +449,15 @@ impl Ord for CStr {
     }
 }
 
+#[stable(feature = "cstr_borrow", since = "1.3.0")]
+impl ToOwned for CStr {
+    type Owned = CString;
+
+    fn to_owned(&self) -> CString {
+        unsafe { CString::from_vec_unchecked(self.to_bytes_with_nul().to_vec()) }
```

Comment:
I think you want `to_bytes` here instead of `to_bytes_with_nul` because the `from_vec_unchecked` function pushes a nul byte on the end.